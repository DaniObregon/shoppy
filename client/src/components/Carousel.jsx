import React, { useEffect, useState } from "react";
import { Box, Flex, Image, Text } from "@chakra-ui/react";

export const Carousel = () => {
  const slides = [
    {
      img: "https://images.pexels.com/photos/6830805/pexels-photo-6830805.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1",
    },
    {
      img: "https://images.pexels.com/photos/9863552/pexels-photo-9863552.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1",
    },
    {
      img: "https://images.pexels.com/photos/3065209/pexels-photo-3065209.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1",
    },
    {
      img: "https://images.pexels.com/photos/7990099/pexels-photo-7990099.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1",
    },
  ];
  const [currentSlide, setCurrentSlide] = useState(0);

  const slidesCount = slides.length;

  const carouselStyle = {
    transition: "all 1.1s",
    ml: `-${currentSlide * 100}%`,
  };

  const SLIDES_INTERVAL_TIME = 3000;
  const ANIMATION_DIRECTION = "right";

  useEffect(() => {
    const prevSlide = () => {
      setCurrentSlide((s) => (s === 0 ? slidesCount - 1 : s - 1));
    };

    const nextSlide = () => {
      setCurrentSlide((s) => (s === slidesCount - 1 ? 0 : s + 1));
    };

    const automatedSlide = setInterval(() => {
      ANIMATION_DIRECTION.toLowerCase() === "left" ? prevSlide() : nextSlide();
    }, SLIDES_INTERVAL_TIME);
    return () => clearInterval(automatedSlide);
  }, [slidesCount]);

  return (
    <Flex
      w="full"
      bg="#edf3f8"
      _dark={{ bg: "#3e3e3e" }}
      p={10}
      alignItems="center"
      justifyContent="center"
    >
      <Flex w="full" overflow="hidden">
        <Flex pos="relative" h="500px" w="full" {...carouselStyle}>
          {slides.map((slide, sid) => (
            <Box key={`slide-${sid}`} flex="none" boxSize="full" shadow="md">
              <Text
                color="white"
                fontSize="xs"
                p="8px 12px"
                pos="absolute"
                top="0"
                whiteSpace="nowrap"
              >
                {sid + 1} / {slidesCount}
              </Text>
              <Image
                src={slide.img}
                alt="carousel image"
                boxSize="full"
                objectFit="cover" // Ajusta la imagen para cubrir el contenedor manteniendo su relaciÃ³n de aspecto
              />
            </Box>
          ))}
        </Flex>
      </Flex>
    </Flex>
  );
};

/** IMAGENES DE ZAPATILLAS
    {
      img: "https://hips.hearstapps.com/hmg-prod/images/run-reebok-running-shoes-64e3beadbe692.jpg",
    },
    {
      img: "https://scontent.faep9-1.fna.fbcdn.net/v/t39.30808-6/221863802_537077584388553_1054731658053590263_n.jpg?_nc_cat=100&ccb=1-7&_nc_sid=127cfc&_nc_eui2=AeEEh83xoHX7uA3M3jybX6IcPs2PM-LdG08-zY8z4t0bTz2WYdy-J1oOS1LypLBSULMMu1nfH3SXI_GrnhXAvLCO&_nc_ohc=ISWDfIVXRyQQ7kNvgGrT7st&_nc_ht=scontent.faep9-1.fna&oh=00_AYC_Go5c-pXVp8JD9ebJjKMzTQxAX-PF20lZ-vedSqFgcQ&oe=66E4BDD4",
    },
    {
      img: "https://brand.assets.adidas.com/capi/enUS/Images/1839569-seo-how-to-break-in-running-shoes-body-image-2_221-681798.jpg",
    },
    {
      img: "https://portcrossfit.com/wp-content/uploads/2023/08/run-nike-running-shoes-646cdd1a19c41-1200x600.jpg",
    },
 */