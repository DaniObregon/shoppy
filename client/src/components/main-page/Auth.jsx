import api from "../../config/axiosConfig"
import React from "react";
import { signInWithPopup, GoogleAuthProvider, signOut } from "firebase/auth";
import { SignUpButton } from "./SignUpButton";
import { SignOutButton } from "./SignOutButton";
import { useNavigate } from "react-router-dom";
import { auth } from "../../config/firebaseConfig";
import { useDispatch, useSelector } from "react-redux";
import { login, logout } from "../../redux/slices/userSlice";
import { Flex, Text, Stack } from "@chakra-ui/react";

export const Auth = ({ onAuthSuccess }) => {
  const provider = new GoogleAuthProvider();
  const navigate = useNavigate();
  const dispatch = useDispatch();

  // Obtiene el estado del usuario desde Redux (obtiene los datos)
  const userInfo = useSelector((state) => state.userInfo);

  const signUpWithGoogle = async () => {
    try {
      const result = await signInWithPopup(auth, provider);
      const user = result.user;
      const token = await user.getIdToken();
      const userData = {
        name: user.displayName,
        email: user.email,
        role_id: 1, // Rol por defecto (client)
        token: token, // Guardamos el token aquí
      };

      //console.log("Datos que se envían al backend:", userData);

      const response = await api.post(
        "/api/auth/signup",
        JSON.stringify(userData),
        {
          headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${token}`, // El token se envia en un header
          },
        }
      );

      //console.log("Respuesta del backend:", response.data);

      // Accede a las propiedades del objeto `user` desde la respuesta del backend
      const { name, email, role } = response.data.user;

      //console.log("Datos recibidos del backend:", { name: name, email: email, role: role });

      // Actualizar el estado global de Redux con los datos del usuario incluyendo el token
      dispatch(
        login({
          name: name,
          email: email,
          role_id: role,
          token: token,
        })
      );

      // Si la autenticación y la creación del usuario son exitosas
      if (onAuthSuccess) {
        onAuthSuccess(user); // Llamar a la función si está definida
      }
      navigate("/"); // Redirige al home después de autenticarse (o cambiar ruta)
    } catch (error) {
      console.error("Error durante el inicio de sesión:", error);
    }
  };

  const signOutUser = async () => {
    try {
      await signOut(auth);
      dispatch(logout()); // Despachar la acción de logout en Redux
      navigate("/");
    } catch (error) {
      console.error("Error al cerrar sesión:", error);
    }
  };

  return (
    <Flex align="center" justify="center" p={4}>
      {userInfo.name ? (
        <Stack direction="row" align="center" spacing={4}>
          <Text
            p={2}
            color="gray.600"
            whiteSpace="nowrap"
            maxWidth="200px"
            overflow="hidden"
            textOverflow="ellipsis"
            fontSize={"m"}
          >
            Hola, {userInfo.name.split(" ")[0]}
          </Text>
          <SignOutButton onClick={signOutUser} />
        </Stack>
      ) : (
        <SignUpButton onClick={signUpWithGoogle} />
      )}
    </Flex>
  );
};
