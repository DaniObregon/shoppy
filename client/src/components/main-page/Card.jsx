import React from "react";
import { useNavigate } from "react-router-dom";
import { chakra, Box, Flex } from "@chakra-ui/react";
import { AddToCartButton } from "./AddToCartButton";

export const Card = ({ id, brand, model, description, price, imgUrl }) => {
  const navigate = useNavigate();

  const handleNavigate = () => {
    navigate(`/product/${id}`);
  };

  return (
    <Flex
      direction="column"
      justifyContent="center"
      alignItems="center"
      w="full"
      mx="auto"
      mb={4}
      transition="transform 0.15s ease-in-out"
      _hover={{
        transform: "scale(1.015)",
        cursor: "pointer",
      }}
      onClick={handleNavigate} // Redirige cuando se hace clic en la card
    >
      <Box
        bg="gray.300"
        h={{ base: 56, md: 64 }} //Imagen m치s grande para mantener la proporci칩n
        w="full"
        rounded="lg"
        shadow="md"
        bgSize="cover"
        bgPos="center"
        style={{
          backgroundImage: `url(${imgUrl})`,
        }}
      ></Box>

      <Box
        w={{ base: "80%", md: "70%" }} // Caja de datos peque침a y centrada
        bg="white"
        _dark={{ bg: "gray.800" }}
        mt={-10} // Superpone la caja de datos en la imagen
        shadow="lg"
        rounded="lg"
        overflow="hidden"
      >
        <chakra.h3
          py={2}
          textAlign="center"
          fontWeight="bold"
          textTransform="uppercase"
          color="gray.800"
          _dark={{ color: "white" }}
          letterSpacing={1}
        >
          {brand}
        </chakra.h3>
        <chakra.p
          py={1}
          textAlign="center"
          fontWeight="medium"
          color="gray.600"
          _dark={{ color: "gray.300" }}
          fontSize="sm"
        >
          {model}
        </chakra.p>
        <chakra.p
          py={1}
          textAlign="center"
          fontWeight="medium"
          color="gray.600"
          _dark={{ color: "gray.300" }}
          fontSize="sm"
          sx={{
            overflow: "hidden",
            display: "-webkit-box",
            WebkitLineClamp: "3",
            WebkitBoxOrient: "vertical",
            wordBreak: "break-word", // Corta palabras largas autom치ticamente
            overflowWrap: "break-word", // Compatibilidad adicional
          }}
        >
          {description}
        </chakra.p>
        <Flex
          alignItems="center"
          justifyContent="space-between"
          py={2}
          px={3}
          bg="gray.200"
          _dark={{ bg: "gray.700" }}
        >
          <chakra.span
            fontWeight="bold"
            color="gray.800"
            _dark={{ color: "gray.200" }}
          >
            ${price}
          </chakra.span>
          <AddToCartButton id={id} />
        </Flex>
      </Box>
    </Flex>
  );
};
