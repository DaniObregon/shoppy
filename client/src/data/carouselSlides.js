export const slides = [
  {
    img: "https://www.brooksrunning.com/on/demandware.static/-/Library-Sites-BrooksRunningShared/default/dw26f51a29/cms-content/Project/ADT/Brooks-Running/EMEA/Blog/2023/April/best-running-shoes-for-knee-pain/S23_RHB_knee_Hero_XL.jpg",
  },
  {
    img: "https://hips.hearstapps.com/hmg-prod/images/run-reebok-running-shoes-64e3beadbe692.jpg",
  },
  {
    img: "https://brand.assets.adidas.com/capi/enUS/Images/1839569-seo-how-to-break-in-running-shoes-body-image-2_221-681798.jpg",
  },
  {
    img: "https://portcrossfit.com/wp-content/uploads/2023/08/run-nike-running-shoes-646cdd1a19c41-1200x600.jpg",
  },
];
