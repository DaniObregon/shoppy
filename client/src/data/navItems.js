import { useNavigate } from "react-router-dom";

export const NAV_ITEMS = (role_id) => {
  const navigate = useNavigate();

  const items = [
    {
      label: "Capilares",
      onClick: () => navigate("/capilares"),
      children: [
        {
          label: "Coloración",
          onClick: () => navigate("/coloracion"),
        },
        {
          label: "Styling",
          onClick: () => navigate("/styling"),
        },
        {
          label: "HairCare",
          onClick: () => navigate("/haircare"),
        },
        {
          label: "Forma",
          onClick: () => navigate("/forma"),
        },
      ],
    },
    {
      label: "Manos y Uñas",
      onClick: () => navigate("/manos-y-unas"),
      children: [
        {
          label: "Prueba children",
          onClick: () => navigate("/coloracion"),
        },
      ],
    },
    {
      label: "Maquillaje",
      onClick: () => navigate("/maquillaje"),
    },
    {
      label: "Equipamientos",
      onClick: () => navigate("/equipamientos"),
    },
    {
      label: "Estética y Spa",
      onClick: () => navigate("/estetica-y-spa"),
    },
  ];

  if (role_id >= 2) {
    items.push({
      label: "Admin Panel",
      onClick: () => navigate("/admin-panel"), // Ruta correcta para el panel de admin
    });
  }
  return items;
};
