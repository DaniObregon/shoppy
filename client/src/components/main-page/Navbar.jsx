import React from "react";
import {
  Box,
  Flex,
  IconButton,
  Stack,
  Collapse,
  useColorModeValue,
  useDisclosure,
} from "@chakra-ui/react";
import { HamburgerIcon, CloseIcon } from "@chakra-ui/icons";
import { Auth } from "./Auth";
import { DarkModeToggle } from "./DarkModeToggle";
import { Logo } from "./Logo";
import { DesktopNav } from "./DesktopNav";
import { MobileNav } from "./MobileNav";

export const Navbar = () => {
  const { isOpen, onToggle } = useDisclosure();

  return (
    <Box mt="60px" marginBottom={1} marginTop={5}>
      <Flex
        bg={useColorModeValue("white", "gray.800")}
        color={useColorModeValue("gray.600", "white")}
        minH={"60px"}
        py={{ base: 2 }}
        px={{ base: 10 }}
        borderBottom={1}
        borderTop={1}
        borderStyle={"solid"}
        borderColor={useColorModeValue("gray.200", "gray.900")}
        align={"center"}
        justify={"space-between"}
        marginTop={8}
        marginBottom={1}
        paddingTop="9px"
        border={-10}
      >
        <Flex
          flex={{ base: 1, md: "auto" }}
          ml={{ base: -2 }}
          display={{ base: "flex", md: "none" }}
        >
          <IconButton
            onClick={onToggle}
            icon={
              isOpen ? <CloseIcon w={3} h={3} /> : <HamburgerIcon w={5} h={5} />
            }
            variant={"ghost"}
            aria-label={"Toggle Navigation"}
          />
        </Flex>
        <Flex flex={{ base: 1 }} justify={{ base: "center", md: "start" }}>
          <Logo />
          <Flex
            display={{ base: "none", md: "flex" }}
            ml={10}
            alignItems={"center"}
          >
            <DesktopNav />
          </Flex>
        </Flex>

        <Stack
          flex={{ base: 1, md: 0 }}
          justify={"flex-end"}
          direction={"row"}
          spacing={6}
          alignItems="center" // Asegura que todo estÃ© alineado verticalmente
        >
          <Auth /> {/* Este componente maneja tanto login como logout */}
          <DarkModeToggle />
        </Stack>
      </Flex>

      <Collapse in={isOpen} animateOpacity>
        <MobileNav />
      </Collapse>
    </Box>
  );
};

<DesktopNav />;
<MobileNav />;
