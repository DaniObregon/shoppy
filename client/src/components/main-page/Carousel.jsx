import React, { useEffect, useState } from "react";
import { Box, Flex, Image, Text } from "@chakra-ui/react";
import { slides } from "../../data/carouselSlides";

export const Carousel = () => {
  const [currentSlide, setCurrentSlide] = useState(0);
  const [visibleSlides, setVisibleSlides] = useState(slides);
  const [isPaused, setIsPaused] = useState(false); // Estado para manejar si la transición está pausada
  const [isHovered, setIsHovered] = useState(false); // Estado para manejar si el mouse está sobre el carrusel

  const slidesCount = visibleSlides.length;
  const carouselStyle = {
    transition: "all 1.5s",
    ml: `-${currentSlide * 100}%`,
  };

  const SLIDES_INTERVAL_TIME = 4000;
  const ANIMATION_DIRECTION = "right";

  useEffect(() => {
    const prevSlide = () => {
      setCurrentSlide((s) => (s === 0 ? slidesCount - 1 : s - 1));
    };

    const nextSlide = () => {
      setCurrentSlide((s) => (s === slidesCount - 1 ? 0 : s + 1));
    };

    const automatedSlide = setInterval(() => {
      if (!isPaused) {
        ANIMATION_DIRECTION.toLowerCase() === "left"
          ? prevSlide()
          : nextSlide();
      }
    }, SLIDES_INTERVAL_TIME);

    return () => clearInterval(automatedSlide);
  }, [slidesCount, isPaused]);

  const handleMouseEnter = () => {
    setIsPaused(true);
    setIsHovered(true);
  };

  const handleMouseLeave = () => {
    setIsPaused(false);
    setIsHovered(false);
  };

  // Función para manejar el error de carga de imágenes y eliminar el slide
  const handleImageError = (index) => {
    setVisibleSlides((prevSlides) =>
      prevSlides.filter((_, slideIndex) => slideIndex !== index)
    );
  };

  return (
    <Flex
      id="carousel"
      role="region"
      aria-label="carousel"
      w="full"
      bg="#edf3f8"
      _dark={{ bg: "#3e3e3e" }}
      p={0}
      alignItems="center"
      justifyContent="center"
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <Flex w="full" overflow="hidden">
        <Flex pos="relative" h="500px" w="full" {...carouselStyle}>
          {visibleSlides.map((slide, index) => (
            <Box
              key={`slide-${index}`}
              flex="none"
              boxSize="full"
              shadow="md"
              transition="transform 0.3s ease"
              style={{
                transform: isHovered ? "translateY(-5px)" : "translateY(0px)",
              }}
            >
              <Text
                color="white"
                fontSize="xs"
                p="8px 12px"
                pos="absolute"
                top="0"
                whiteSpace="nowrap"
              >
                {index + 1} / {slidesCount}
              </Text>
              <Image
                src={slide.img}
                alt={`Slide ${index + 1}`}
                boxSize="full"
                objectFit="cover"
                style={{
                  opacity: isHovered ? "0.8" : "1",
                  transition: "opacity 0.3s ease",
                }}
                onError={() => handleImageError(index)} // Maneja el error de carga de imagen eliminando el slide
              />
            </Box>
          ))}
        </Flex>
      </Flex>
    </Flex>
  );
};
