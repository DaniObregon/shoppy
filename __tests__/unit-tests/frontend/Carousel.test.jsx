import "@testing-library/jest-dom";
import React from "react";
import { render, screen, act, fireEvent } from "@testing-library/react";
import { Carousel } from "../../../client/src/components/main-page/Carousel";
import { ChakraProvider } from "@chakra-ui/react";
import { slides } from "../../../client/src/data/carouselSlides"; // Importamos los slides directamente

describe("Carousel component", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllTimers();
  });

  test("should render the carousel with the correct number of slides", () => {
    render(
      <ChakraProvider>
        <Carousel />
      </ChakraProvider>
    );

    const slidesCount = slides.length;
    const images = screen.getAllByAltText(/Slide \d+/i);

    // Verificar que se renderizan todas las imágenes de los slides
    expect(images).toHaveLength(slidesCount);

    // Verificar que la numeración se muestra correctamente (ej. "1 / 4" para el primer slide)
    expect(screen.getByText(`1 / ${slidesCount}`)).toBeInTheDocument();
  });

  test("should automatically transition to the next slide after the specified interval", () => {
    render(
      <ChakraProvider>
        <Carousel />
      </ChakraProvider>
    );

    const slidesCount = slides.length;
    const intervalTime = 3000; // Tiempo de intervalo

    // Verificar que el primer slide se muestra inicialmente
    expect(screen.getByText(`1 / ${slidesCount}`)).toBeInTheDocument();

    // Avanzar el tiempo dentro de un bloque act para forzar el ciclo de renderizado
    act(() => {
      jest.advanceTimersByTime(intervalTime);
    });
    expect(screen.getByText(`2 / ${slidesCount}`)).toBeInTheDocument();

    act(() => {
      jest.advanceTimersByTime(intervalTime);
    });
    expect(screen.getByText(`3 / ${slidesCount}`)).toBeInTheDocument();

    // Avanzar tiempo hasta que vuelva al primer slide (suponiendo que es cíclico)
    act(() => {
      for (let i = 3; i <= slidesCount; i++) {
        jest.advanceTimersByTime(intervalTime);
      }
    });
    expect(screen.getByText(`1 / ${slidesCount}`)).toBeInTheDocument();
  });

  test("should match snapshot", () => {
    const { asFragment } = render(
      <ChakraProvider>
        <Carousel />
      </ChakraProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should pause the auto-transition when the user hovers over the carousel", () => {
    render(
      <ChakraProvider>
        <Carousel />
      </ChakraProvider>
    );

    const intervalTime = 3000;
    const carousel = screen.getByRole("region", { name: /carousel/i });

    // Simular el evento de hover
    fireEvent.mouseEnter(carousel);

    // Avanzar el tiempo pero asegurar que el slide no cambie debido al hover
    jest.advanceTimersByTime(intervalTime);
    expect(screen.getByText(`1 / ${slides.length}`)).toBeInTheDocument();
  });

  test("should resume the auto-transition when the user stops hovering over the carousel", () => {
    jest.useFakeTimers(); // Usa temporizadores simulados
    render(
      <ChakraProvider>
        <Carousel />
      </ChakraProvider>
    );

    const intervalTime = 3000; // Tiempo de intervalo
    const carousel = screen.getByRole("region", { name: /carousel/i });

    // Simular el evento de hover para pausar
    fireEvent.mouseEnter(carousel);

    // Simular la salida del hover para reanudar
    fireEvent.mouseLeave(carousel);

    // Avanzar el temporizador y verificar que el slide cambia
    jest.advanceTimersByTime(intervalTime);
    expect(screen.getByText(`2 / ${slides.length}`)).toBeInTheDocument(); // Verifica que el siguiente slide esté visible
  });

  test("should remove a slide when its image fails to load", () => {
    render(
      <ChakraProvider>
        <Carousel />
      </ChakraProvider>
    );

    const initialSlidesCount = slides.length;
    const imageWithError = screen.getAllByRole("img")[1]; // Simula el error en la segunda imagen

    // Disparar el evento de error en la carga de la imagen
    fireEvent.error(imageWithError);

    // Verificar que el número de slides visibles se haya reducido en 1
    expect(screen.getAllByRole("img")).toHaveLength(initialSlidesCount - 1);

    // Verificar que la numeración se actualizó correctamente
    expect(
      screen.getByText(`1 / ${initialSlidesCount - 1}`)
    ).toBeInTheDocument();
  });
});
