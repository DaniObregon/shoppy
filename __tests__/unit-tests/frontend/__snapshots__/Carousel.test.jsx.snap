// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="carousel"
    class="css-1uskzmk"
    id="carousel"
    role="region"
  >
    <div
      class="css-1r09swm"
    >
      <div
        class="css-jolvvj"
      >
        <div
          class="css-1egvv0c"
          style="transform: translateY(0px);"
        >
          <p
            class="chakra-text css-p5m561"
          >
            1 / 4
          </p>
          <img
            alt="Slide 1"
            class="chakra-image css-1ekmhiq"
            src="https://www.brooksrunning.com/on/demandware.static/-/Library-Sites-BrooksRunningShared/default/dw26f51a29/cms-content/Project/ADT/Brooks-Running/EMEA/Blog/2023/April/best-running-shoes-for-knee-pain/S23_RHB_knee_Hero_XL.jpg"
            style="opacity: 1; transition: opacity 0.3s ease;"
          />
        </div>
        <div
          class="css-1egvv0c"
          style="transform: translateY(0px);"
        >
          <p
            class="chakra-text css-p5m561"
          >
            2 / 4
          </p>
          <img
            alt="Slide 2"
            class="chakra-image css-1ekmhiq"
            src="https://hips.hearstapps.com/hmg-prod/images/run-reebok-running-shoes-64e3beadbe692.jpg"
            style="opacity: 1; transition: opacity 0.3s ease;"
          />
        </div>
        <div
          class="css-1egvv0c"
          style="transform: translateY(0px);"
        >
          <p
            class="chakra-text css-p5m561"
          >
            3 / 4
          </p>
          <img
            alt="Slide 3"
            class="chakra-image css-1ekmhiq"
            src="https://brand.assets.adidas.com/capi/enUS/Images/1839569-seo-how-to-break-in-running-shoes-body-image-2_221-681798.jpg"
            style="opacity: 1; transition: opacity 0.3s ease;"
          />
        </div>
        <div
          class="css-1egvv0c"
          style="transform: translateY(0px);"
        >
          <p
            class="chakra-text css-p5m561"
          >
            4 / 4
          </p>
          <img
            alt="Slide 4"
            class="chakra-image css-1ekmhiq"
            src="https://portcrossfit.com/wp-content/uploads/2023/08/run-nike-running-shoes-646cdd1a19c41-1200x600.jpg"
            style="opacity: 1; transition: opacity 0.3s ease;"
          />
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
